### PR TITLE
[tests-only]tests: add tests for removing share permission

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
https://owncloud.dev/libre-graph-api/#/drives.permissions/DeletePermission
Scenarios added:
1. Scenario Outline: user removes access from a resource in the user share
2. Scenario Outline: user removes access from a resource inside of a project space in the user share

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/8008

## Motivation and Context
To provide test coverage for `/v1beta1/drives/{drive-id}/items/{item-id}/permissions/{perm-id}` 

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+    And using spaces DAV path
+
+
+  Scenario Outline: user removes access from a folder
+    Given user "Alice" has created folder "folderToShare"
+    And user "Alice" has shared folder "folderToShare" with user "Brian" with permissions "<permissions>"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:14]: this invite probably needs to be sent with sharingNG api. I'll update it

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      | Co Owner    | file          | textfile.txt  |
+      | Manager     | file          | textfile.txt  |
+      | Viewer      | folder        | FolderToShare |
+      | Editor      | folder        | FolderToShare |
+      | Co Owner    | folder        | FolderToShare |
+      | Uploader    | folder        | FolderToShare |
+      | Manager     | folder        | FolderToShare |
+
+
+  Scenario Outline: user removes access from a file inside of a project space
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:39]:
```suggestion
  Scenario Outline: user removes access from a resource inside of a project space
```

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 3]
```diff
@@ -0,0 +1,65 @@
+Feature: Delete access to a drive item
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:1]:
```suggestion
Feature: Remove access to a drive item
```

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+    And using spaces DAV path
+
+
+  Scenario Outline: user removes access from a resource in the user share
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:12]: is it delete/revoke share?

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 5]
```diff
@@ -84,15 +85,16 @@ public function theUserPermissionsListOfResource(string $user, string $fileOrFol
...
 	 *
-	 * @return void
+	 * @return ResponseInterface
+	 *
+	 * @throws JsonException
+	 * @throws \GuzzleHttp\Exception\GuzzleException
 	 * @throws Exception
 	 */
-	public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $user, TableNode $table): void {
+	public function sendSharingInviteUsingGraphAPI(string $user, TableNode $table): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:97]:
```suggestion
	public function sendSharingInvitation(string $user, TableNode $table): ResponseInterface {
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 6]
```diff
@@ -108,20 +110,49 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
...
+	 * @Given /^user "([^"]*)" has sent the following share invitation:$/
+	 *
+	 * @param string $user
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 * @throws \GuzzleHttp\Exception\GuzzleException
+	 */
+	public function userHasSentShareInvitation(string $user, TableNode $table): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:138]:
```suggestion
	public function userHasSentTheFollowingShareInvitation(string $user, TableNode $table): void {
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 7]
```diff
@@ -169,4 +200,45 @@ public function userCreatesAPublicLinkShareWithSettings(string $user, TableNode
...
+	 * @param string $shareeUser
+	 * @param string $resourceType
+	 * @param string $resource
+	 * @param string $space
+	 *
+	 * @return void
+	 * @throws JsonException
+	 * @throws \GuzzleHttp\Exception\GuzzleException
+	 */
+	public function removeSharePermissionOfAResourceUsingGraphAPI(string $user, string $shareeUser, string $resourceType, string $resource, string $space): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:217]:
```suggestion
	public function userRemovesSharePermissionOfUserFromResourceOfSpaceUsingGraphAPI(string $user, string $shareeUser, string $resourceType, string $resource, string $space): void {
```

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature (hunk 4 above)
Comment (SwikritiT) [reply tests/acceptance/features/apiSharingNg/deletePermissions.feature:12]: I'm using the exact wording from swagger https://owncloud.dev/libre-graph-api/#/drives.permissions/DeletePermission
it says `Remove access to a DriveItem.`  but in practicality yes it deletes the share
```suggestion
  Scenario Outline: user removes access to a resource in the user share
```

Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/deletePermissions.feature:12]: Noice!

**Review by saw-jan — APPROVED** [review]

**Pull request merged**